src/parser/mod.rs

```diff
@@ -6722,6 +6726,8 @@ impl<'a> Parser<'a> {
             SetExpr::Values(self.parse_values(is_mysql)?)
         } else if self.parse_keyword(Keyword::TABLE) {
             SetExpr::Table(Box::new(self.parse_as_table()?))
+        } else if self.parse_keyword(Keyword::FROM) {
```

Comment:
I have no idea why, but adding this branch causes SO in the recursion tests such as `parse_deeply_nested_parens_hits_recursion_limits`

Reply:
even if the branch is never met, the presence of it is causing SO.

Reply:
maybe @alamb, @andygrove @nickolay knows why adding this condition causes a SO.

Reply:
This branch probably makes the stack frame size increase in debug builds as it adds some new local variables. I think we could fix the test by decreasing the recursion limit